### ISSUE TITLE
[FEAT] New cards first?

### ISSUE THREAD

**Issue by csss1234** [st3v3nmw/obsidian-spaced-repetition#610]
**Is your feature request related to a problem? Please describe.**
I have about 100 cards in one deck, and I add new ones regularly. During review, I only process about 20-30. When I click "review flashcards from notes" (regardless of "random order" setting), I don't get the new ones first. So I could review my 100 cards 10 times, and never see the new ones. 

**Describe the solution you'd like**
Option to show new/unrated first, or to choose "just the new ones"

**Comment by maggaou** [comment]
I'm not sure how to reproduce that. I haven't had that problem. My deck has 210 cards and in every review I have got due cards (green) and new cards (blue).

**Comment by csss1234** [comment]
> I'm not sure how to reproduce that. I haven't had that problem. My deck has 210 cards and in every review I have got due cards (green) and new cards (blue).

Yes I see the green blue distinction. But still when I choose the deck, I'm always presented with the Dues before the News. What do you get served first?

**Comment by maggaou** [comment]
> Yes I see the green blue distinction. But still when I choose the deck, I'm always presented with the Dues before the News. What do you get served first?

I think new cards come after due cards. But this isn't problem since I always review everything. For example now I have 19+4 cards to review.

**Comment by csss1234** [comment]
> > Yes I see the green blue distinction. But still when I choose the deck, I'm always presented with the Dues before the News. What do you get served first?
> 
> I think new cards come after due cards. But this isn't problem since I always review everything. For example now I have 19+4 cards to review.

Right, so we are seeing the same thing. I only review about 20-30 of my 100+ cards each time. And I like very short repeat days, so I well often set them to be "due" tomorrow. I want to review them daily. 

Since due comes before new, I just see the same 20 over and over, even with the "randomize" setting on.

**Comment by maggaou** [comment]
Ok I see. The plugin should randomize green and blue cards together and not separately.

Lähetetty sovelluksesta Outlook for iOS<https://aka.ms/o0ukef>
________________________________
Lähettäjä: csss1234 ***@***.***>
Lähetetty: Saturday, January 28, 2023 5:35:52 PM
Vastaanottaja: st3v3nmw/obsidian-spaced-repetition ***@***.***>
Kopio: maggaou ***@***.***>; Comment ***@***.***>
Aihe: Re: [st3v3nmw/obsidian-spaced-repetition] [FEAT] New cards first? (Issue #610)


Yes I see the green blue distinction. But still when I choose the deck, I'm always presented with the Dues before the News. What do you get served first?

I think new cards come after due cards. But this isn't problem since I always review everything. For example now I have 19+4 cards to review.

Right, so we are seeing the same thing. I only review about 20-30 of my 100+ cards each time. And I like very short repeat days, so I well often set them to be "due" tomorrow. I want to review them daily.

Since due comes before new, I just see the same 20 over and over, even with the "randomize" setting on.

**Comment by csss1234** [comment]
> Ok I see. The plugin should randomize green and blue cards together and not separately. Lähetetty sovelluksesta Outlook for iOS<https://aka.ms/o0ukef>
> […](#)

I'd like even more to have an option to show new first.

**Comment by ronzulu** [comment]
Hi @csss1234 I have enhanced the plug-in to give greater flexibility in how cards are ordered during review.

Would be great if you (and anyone else) could Beta Test. There are two new options that replace the existing single "Randomize card order during review" option.

![image](https://github.com/st3v3nmw/obsidian-spaced-repetition/assets/75528127/6821488f-7be8-4794-ad77-bac47e72bcdf)

I've been using the "Random card from random deck" option which I quite like.

![image](https://github.com/st3v3nmw/obsidian-spaced-repetition/assets/75528127/d2793128-dba0-467b-8fb7-a0a51325545c)

There is a file called `main.js` in this zip file 
[main.zip](https://github.com/st3v3nmw/obsidian-spaced-repetition/files/13256794/main.zip)

You will need to copy that to the obsidian plug-in folder, e.g.
`D:\Obsidian\Obsidian\.obsidian\plugins\obsidian-spaced-repetition`

Then reload the plug-in.

Cheers
Ronny

**Comment by moiSentineL** [comment]
> Would be great if you (and anyone else) could Beta Test

Thanks, I tried it. It is working, but there's one issue. 
If there is a LAST due card of a deck for the review, once it is marked, the GUI returns to the menu displaying the decks.
This doesn't happen in the case of other cards. 

This resembles how, after reviewing the whole deck, it returns to the home menu.

Like shown in this [video](https://github.com/st3v3nmw/obsidian-spaced-repetition/assets/68242099/2a5549c6-365a-4276-bae3-f614bbd39900)

**Comment by ronzulu** [comment]
Hi @moiSentineL 

I think I've fixed this now. When you have a chance, could you please test and let me know the result.

[main_610A.zip](https://github.com/st3v3nmw/obsidian-spaced-repetition/files/13499665/main_610A.zip)

Cheers
Ronny

**Comment by moiSentineL** [comment]
Thanks, @ronzulu
It works fine now. There has been no instances of that bug after 15-20 mins of doing the flashcards.

**Comment by ronzulu** [comment]
Hi @moiSentineL  the has been included in the latest release 1.11.0. Please close this at your convenience.

Cheers
Ronny

**Comment by 4Source** [comment]
I think it could be nice if the Options get split up. 

### Order the cards in a deck are displayed during review
- Sequentially 
- Randomly 

### Order priority of cards in a deck
- Newest first
- Due first

### Order decks are displayed during review
- Sequentially 
- Randomly

### Switch decks during review
- Never
- Deck finished
- Always

**Comment by 4Source** [comment]
Not sure this is allready fixed or I just have an different point of view. But for me in the selection of the decks when you click on lowest level of decks (1) this is the deck this makes sense. But if you click on one of the parents decks (2) this not the deck instead it switches between the subdecks. 

I think it would make much more sense if the "clicked deck" is the deck and containing the cards of its child decks.
<img width="597" alt="image" src="https://github.com/st3v3nmw/obsidian-spaced-repetition/assets/38220764/4f362af7-b689-4e80-8c85-a4bbd1fa2de8">
